core/src/main/kotlin/xtdb/Types.kt

```diff
@@ -112,8 +113,17 @@ fun valueToArrowType(obj: Any?) = when (obj) {
     is Float -> MinorType.FLOAT4.type
     is Double -> MinorType.FLOAT8.type
 
-    // HACK we should support parameterised decimals here
-    is BigDecimal -> ArrowType.Decimal(38, 19, 128)
+    is BigDecimal -> {
+        val precision = when (obj.precision()) {
+            // Java Arrow only supports 128 and 256 bit widths
+//            in 0..7 -> 32
+//            in 8..16 -> 64
+            in 0..32 -> 32
+            in 33..64 -> 64
+            else -> throw IllegalArgumentException("unsupported precision: ${obj.precision()}")
```

Comment:
@jarohen FYI. There is a range of values we are representing here
```clj
  (def bd 115792089237316195423570985008687907853269984665640564039457584007913129639935M)
  
  (-> (.unscaledValue bd) (.bitLength))
  ;;    256

  (count (str bd))
  ;; => 78
```

Reply:
I think I'll cope with that :sweat_smile: IIUC it's an Arrow restriction? if so, not a lot we're likely to be able to do about it unless we create extension vectors.

Let's wait for the first user with those precision requirements before implementing it :)

Reply:
Sorry, if I wasn't clear. We could replace the second `in` with `33..77` and make it precision 77 and that would I think "just work" as it fits a `Decimal256Vector`.